tests/cypress/tests/unit/single_column_syntactic_tree_test.ts

```diff
@@ -110,9 +117,119 @@ describe('Single Column Syntax Tree', () => {
         const tree = new SingleColumnSyntaxTree('"1"', COLUMN_TEXT);
 
         expect(tree.isValid).to.equal(true);
+        expect(tree.evaluate({ a: 1 })).to.equal(true);
+        expect(tree.evaluate({ a: 2 })).to.equal(false);
         expect(tree.evaluate({ a: '1' })).to.equal(true);
         expect(tree.evaluate({ a: '2' })).to.equal(false);
 
         expect(tree.toQueryString()).to.equal('{a} contains "1"');
     });
+
+    ['1975', '"1975"'].forEach(value => {
+        it(`can be expression '${value}' with datetime column type`, () => {
+            const tree = new SingleColumnSyntaxTree(value, COLUMN_DATE);
+
+            expect(tree.evaluate({ a: 1975 })).to.equal(true);
+            expect(tree.evaluate({ a: '1975' })).to.equal(true);
+            expect(tree.evaluate({ a: '1975-01' })).to.equal(true);
+            expect(tree.evaluate({ a: '1975-01-01' })).to.equal(true);
+            expect(tree.evaluate({ a: '1975-01-01 01:01:01' })).to.equal(true);
+
+            expect(tree.evaluate({ a: 1976 })).to.equal(false);
+            expect(tree.evaluate({ a: '1976' })).to.equal(false);
+            expect(tree.evaluate({ a: '1976-01' })).to.equal(false);
+            expect(tree.evaluate({ a: '1976-01-01' })).to.equal(false);
+            expect(tree.evaluate({ a: '1976-01-01 01:01:01' })).to.equal(false);
+        });
+    });
+
+    [
+        { type: COLUMN_UNDEFINED, name: 'undefined' },
+        { type: COLUMN_ANY, name: 'any' },
+        { type: COLUMN_TEXT, name: 'text' }
+    ].forEach(({ type, name }) => {
+        it(`returns the correct relational operator lexeme for '${name}' column type`, () => {
+            const tree = new SingleColumnSyntaxTree('1', type);
+            const structure = tree.toStructure();
+
+            expect(tree.toQueryString()).to.equal('{a} contains 1');
+            expect(structure).to.not.equal(null);
+
+            if (structure) {
+                expect(structure.value).to.equal(RelationalOperator.Contains);
+                expect(structure.subType).to.equal(RelationalOperator.Contains);
+                expect(structure.type).to.equal(LexemeType.RelationalOperator);
+
+                expect(structure.left).to.not.equal(null);
+                if (structure.left) {
+                    expect(structure.left.type).to.equal(LexemeType.Expression);
+                    expect(structure.left.subType).to.equal('field');
+                    expect(structure.left.value).to.equal('a');
+                }
+
+                expect(structure.right).to.not.equal(null);
+                if (structure.right) {
+                    expect(structure.right.type).to.equal(LexemeType.Expression);
+                    expect(structure.right.subType).to.equal('value');
+                    expect(structure.right.value).to.equal(1);
+                }
+            }
+        });
+    });
```

Comment:
Without the change to the single column syntax tree
https://circleci.com/gh/plotly/dash-table/13136, https://github.com/plotly/dash-table/pull/589/commits/04bbe1d41f30cfd1e634c6274405943db89e7fcf
and passing again afterwards
https://circleci.com/gh/plotly/dash-table/13143, https://github.com/plotly/dash-table/pull/589/commits/8087b40d7d96d6147ae45decd5f6a14f082efc67